erts/etc/win32/erl.c

```diff
@@ -235,33 +235,51 @@ static wchar_t *find_erlexec_dir(wchar_t *erlpath)
     wchar_t *dir =_wcsdup(erlpath);
     wchar_t *p;
     wchar_t *p2;
+
+    wchar_t *long_dir;
+    DWORD length;
     
     /* Chop of base name*/
     for (p = dir+wcslen(dir)-1 ;p >= dir && *p != L'\\'; --p)
         ;
     *p =L'\0';
     p--;
 
+    length = GetLongPathNameW(dir, NULL, 0);
+    if(length == 0) {
+        free(dir);
+        error("Cannot find erlexec.dll");
+    }
+    long_dir = (wchar_t *) malloc(length * sizeof(wchar_t));
+    if(GetLongPathNameW(dir, long_dir, length) == 0) {
+        free(dir);
+        free(long_dir);
+        error("Cannot find erlexec.dll");
+    }
+    p = long_dir + wcslen(long_dir) - 1;
```

Comment:
We already have `length`, there's no need to determine it again.

Reply:
I think we can decuce it from what was returned by GetLongPathNameW, I will fix that.